### DISCUSSION TITLE
How to share plugins across projects?

### DISCUSSION THREAD

**Discussion by quant-daddy** [dotansimha/graphql-code-generator#6155] [Q&A]
We are using graphql codegen across multiple projects. We want the plugin packages to be consistent across these projects. One way is to have another package which has all the plugins along with cli plugin installed as its dependency (typescript, typescript-operations etc) and share this package across the projects. However the cli requires all the plugin packages to be installed in the root node_modules of the project. Is there a way to make the cli reference the plugins installed in the same package where the cli is installed?

**Comment by dotansimha** [comment]
Hi @skk2142 ! 
This is a great question. I think if I'm trying to break it down, we have 2 (maybe separate?) issues here:

1 - How to share codegen configuration across different projects/repos/workspaces?

I think the ideal way to do that is to use an internal package that depends on codegen and all the relevant plugins, and also comes with the config file of codegne. You can use `.js` extension for the config to make it easier to build and export. Then, projects can run `codegen --config node_modules/@myorg/codegen`. 

2 - How to deal with different levels of `node_modules`? 

I think it should be possible to set a path directly for loading plugins, and then the shared config file (preferably `.js`) you are creating can use __dirname or even require on it's own, and this might work.
